### PR TITLE
PMM-6897 Add Suspend/Resume psmdb

### PR DESCRIPTION
**Pull request by askomorokhov**
[PMM-6897](https://jira.percona.com/browse/PMM-6897)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/0

https://github.com/percona-platform/dbaas-controller/pull/100
https://github.com/percona-platform/dbaas-api/pull/69
https://github.com/percona/pmm/pull/634

---
- [x] Tests passed.
- [x] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [x] Update API dependency.
- [x] Update jira ticket description if needed.
- [x] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by percona-csalguero — APPROVED** [review]

File: services/management/dbaas/psmdb_cluster_service.go [hunk 1]
```diff
@@ -86,6 +86,10 @@ func (s PSMDBClusterService) ListPSMDBClusters(ctx context.Context, req *dbaasv1
 			State: psmdbStates()[c.State],
 		}
 
+		if c.Params.Paused {
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:89]: Until it's really paused I think status should be CHANGING

File: services/management/dbaas/psmdb_cluster_service.go [hunk 2]
```diff
@@ -169,17 +173,32 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
...
-					CpuM:        req.Params.Replicaset.ComputeResources.CpuM,
-					MemoryBytes: req.Params.Replicaset.ComputeResources.MemoryBytes,
-				},
-			},
-		},
 	}
 
+	if req.Params != nil {
+		in.Params = new(dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams)
+		in.Params.ClusterSize = req.Params.ClusterSize
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:180]:
```suggestion
		in.Params = &dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams {
		    ClusterSize: req.Params.ClusterSize,
		}
```

File: services/management/dbaas/psmdb_cluster_service.go [hunk 3]
```diff
@@ -169,17 +173,32 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
...
+
+		if req.Params.Suspend && req.Params.Resume {
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:182]: I think it should be higher before initializing `in.Params`

File: services/management/dbaas/psmdb_cluster_service.go [hunk 4]
```diff
@@ -169,17 +173,32 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
...
+		in.Params.Suspend = req.Params.Suspend
+		in.Params.Resume = req.Params.Resume
+
+		if req.Params.Replicaset != nil {
+			in.Params.Replicaset = new(dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams_ReplicaSet)
+		}
+
+		if req.Params.Replicaset.ComputeResources != nil {
+			in.Params.Replicaset.ComputeResources = new(dbaascontrollerv1beta1.ComputeResources)
+			if req.Params.Replicaset.ComputeResources.CpuM > 0 {
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:194]: Do we really need this `if`?

File: services/management/dbaas/psmdb_cluster_service.go [hunk 5]
```diff
@@ -169,17 +173,32 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
...
+				in.Params.Replicaset.ComputeResources.CpuM = req.Params.Replicaset.ComputeResources.CpuM
+			}
+			if req.Params.Replicaset.ComputeResources.MemoryBytes > 0 {
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:197]: and this?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: services/management/dbaas/psmdb_cluster_service.go (hunk 1 above)
Comment (BupycHuk) [reply services/management/dbaas/psmdb_cluster_service.go:89]: so I think we should check if paused and cluster is ready.

Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: Fixed.

File: services/management/dbaas/psmdb_cluster_service.go (hunk 2 above)
Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: Changed.

File: services/management/dbaas/psmdb_cluster_service.go (hunk 3 above)
Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: Moved.

File: services/management/dbaas/psmdb_cluster_service.go (hunk 4 above)
Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: Yeap, we do not it.

File: services/management/dbaas/psmdb_cluster_service.go (hunk 5 above)
Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: fixed.

File: services/management/dbaas/psmdb_cluster_service.go [hunk 6]
```diff
@@ -169,17 +173,31 @@ func (s PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *dbaasv
...
+		}
+
+		in.Params.Suspend = req.Params.Suspend
+		in.Params.Resume = req.Params.Resume
+
+		if req.Params.Replicaset != nil {
+			in.Params.Replicaset = new(dbaascontrollerv1beta1.UpdatePSMDBClusterRequest_UpdatePSMDBClusterParams_ReplicaSet)
+		}
+
+		if req.Params.Replicaset.ComputeResources != nil {
```
Comment (BupycHuk) [inline services/management/dbaas/psmdb_cluster_service.go:194]: it might raise an error if Replicaset is nil

Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go]: Yes, correct. Fixed.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**